common/test/mock-ingester/src/lib.rs

```diff
@@ -0,0 +1,279 @@
+        Ok(()).into()
+    }
+
+    fn call(&mut self, req: Request<Body>) -> Self::Future {
+        info!("Received {:?}", req);
+        let files = self.files.clone();
+        Box::pin(async move {
+            let rsp = Response::builder();
+
+            let uri = req.uri();
+            if uri.path() != ROOT {
+                return Ok(rsp.status(404).body(Body::empty()).unwrap());
+            }
+
+            let encoding = {
+                &req.headers()
+                    .get("content-encoding")
+                    .map(|e| e.to_str().unwrap().to_string())
+                    .clone()
+            };
+
+            let mut bytes = Vec::new();
+            let mut body = req.into_body();
+            match encoding {
+                Some(encoding) if encoding == "gzip" => {
+                    let mut decoder = async_compression::stream::GzipDecoder::new(
+                        body.map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e)),
+                    );
+                    while let Some(Ok(chunk)) = decoder.next().await {
+                        bytes.extend_from_slice(&chunk);
+                    }
+                }
+                _ => {
+                    while let Some(Ok(chunk)) = body.next().await {
+                        bytes.extend_from_slice(&chunk);
+                    }
+                }
+            }
+
+            let ingest_body: IngestBody = match serde_json::from_slice(&bytes) {
+                Ok(lines) => lines,
+                Err(e) => {
+                    error!("{}", e);
+                    return Ok(rsp.status(500).body(Body::empty()).unwrap());
+                }
+            };
+
+            for line in ingest_body.lines {
+                if let Some(mut raw_line) = line.line {
+                    if !raw_line.ends_with('\n') {
+                        raw_line.push('\n')
+                    }
+
+                    let orig_file_name = line.file.unwrap_or_else(|| " unknown".into());
+
+                    let file_name = orig_file_name.replace("/", "-").clone();
+                    let file_name = &file_name[1..];
+
+                    let mut files = files.lock().await;
+                    let (ref line_count, ref mut file) =
+                        files.entry(orig_file_name).or_insert_with(|| {
+                            info!("creating {}", file_name);
+                            (
+                                AtomicUsize::new(0),
+                                OpenOptions::new()
+                                    .create(true)
+                                    .append(true)
+                                    .open(&file_name)
+                                    .unwrap(),
+                            )
+                        });
+
+                    line_count.fetch_add(1, Ordering::Relaxed);
+                    file.write_all(raw_line.as_bytes()).unwrap();
```

Comment:
What's the reason to use a file? should we store it in memory instead?

We could use some kind of mutex and just append in memory (just a few kbs/mbs).

Reply:
The file logic was lifter directly from the original mock ingester code, we could quite easily discard the content for now, or simply append to a Vec of Bytes, any thoughts?